### PR TITLE
feat(post): exclude articles with future publication date

### LINKED ISSUE #351: Articles with a future publication date should never be visible
Create new folder under `/en/publish` with a futuristic date
Create a new post entry in that folder
Reload the homepage
-> the post entry appears on the homepage

Future posts should never appear.

Since we only store "dates" (YYYY/MM/DD) with no time, we should at least make sure that a surfer cannot see a post with DD+1, independently of the time. Which also means that people in Japan will be able to access the content before the people in the US.

### PR DESCRIPTION
**Pull request by kptdobe**
Fix #351

### CODE REVIEW

**Comment by kptdobe** [comment]
Not sure if there isn't an easier way to handle that.

File: scripts/common.js [hunk 1]
```diff
@@ -320,7 +320,15 @@ export async function fetchArticleIndex(offset) {
 
   if (response.ok) { 
     const json = await response.json();
-    const data = Array.isArray(json) ? json : json.data;
+    let data = Array.isArray(json) ? json : json.data;
+
+    // filter out future date
+    const current = new Date();
+    const today = new Date(Date.UTC(current.getFullYear(), current.getMonth(), current.getDate()));
```
Comment (tripodsan) [inline scripts/common.js:327]: why UTC ?

Comment (tripodsan) [reply scripts/common.js:327]: simpler:

```
today.setHours(23);
const eod = today.setMinutes(59, 59, 999);
```

File: scripts/common.js [hunk 2]
```diff
@@ -320,7 +320,15 @@ export async function fetchArticleIndex(offset) {
 
   if (response.ok) { 
     const json = await response.json();
-    const data = Array.isArray(json) ? json : json.data;
+    let data = Array.isArray(json) ? json : json.data;
+
+    // filter out future date
+    const current = new Date();
```
Comment (tripodsan) [inline scripts/common.js:326]: simpler:
```
const today = new Date();
today.setHours(23);
const eod = today.setMinutes(59, 59, 999);
```

**Comment by kptdobe** [comment]
I spent hours to try to setup Mocha and Babel to write a test but it does not want to transpile the `scripts/*.js` for unknown reason. I will need more time to do the setup. I'll add this test later when I will manage the setup.

**Comment by rofe** [comment]
One caveat of not doing this via query and in Excel is that users will be able to find the URLs to the future posts in the JSON request. Another is that if there are less than 12 posts left, the first deck page won't be filled completely...

But I think that's ok for a temporary solution until we know precisely at what time posts should become available.

**Comment by kptdobe** [comment]
I just had a conversation with @davidnuescheler about it, we'll add the filtering on the Excel side too. Like this, no data will leak and in case they do, at least it does not get displayed.

**Pull request merged**